### PR TITLE
#3549 DataTable: (inline-)edit - row-mode: set focus to...

### PR DESCRIPTION
**Pull request by christophs78**
… the first input-field in the row

### CODE REVIEW

File: src/main/resources/META-INF/resources/primefaces/datatable/datatable.js [hunk 1]
```diff
@@ -2399,6 +2399,12 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
         else {
             this.showRowEditors(row);
 
+            let inputs=$(row).find('input, select, textarea, button');
```
Comment (melloware) [inline src/main/resources/META-INF/resources/primefaces/datatable/datatable.js]: I think a better selector would be `:input:enabled` which finds the first input,select,textarea radio button or any input element that is actually NOT disabled.

File: src/main/resources/META-INF/resources/primefaces/datatable/datatable.js [hunk 2]
```diff
@@ -2399,6 +2399,12 @@ PrimeFaces.widget.DataTable = PrimeFaces.widget.DeferredWidget.extend({
...
+            if (inputs.length > 0) {
+                let firstInput = inputs.first();
+                firstInput.focus();
```
Comment (melloware) [inline src/main/resources/META-INF/resources/primefaces/datatable/datatable.js]: I think you should just make this one line with `inputs.first().focus();` no need to declare variable for this yet.

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by tandraschko** [comment]
let is supported in IE?

**Comment by melloware** [comment]
It looks like Let is supported in IE11 https://stackoverflow.com/a/2356926/502366

**Comment by christophs78** [comment]
> 
> 
> It looks like Let is supported in IE11 https://stackoverflow.com/a/2356926/502366

Yep. :-) https://caniuse.com/#feat=let

**Comment by christophs78** [comment]
@melloware : Thanks for your feedback. Done.

**Comment by mertsincan** [comment]
Hi @christophs78,

Thanks a lot for the PR! But, I think you need to make these changes in showRowEditors. Because, row editor has lazy feature. 
```
      if(this.cfg.rowEditMode === "lazy") {
            this.lazyRowEditInit(row);
        }
```
When users use this feature, your code won't work for them.

Also, could you please use 'var' instead of 'let'? We don't use ES6 syntax in PrimeFaces.

Maybe, 
```
showRowEditors: function(row) {
        var isFirstItemFocused = false;
        row.addClass('ui-state-highlight ui-row-editing').children('td.ui-editable-column').each(function() {
            var column = $(this);
    /** Maybe **/
            if (!isFirstItemFocused) {
               var inputs = column.find('...');
               if (inputs.length > 0) {
                    ...
                    isFirstItemFocused = true;
                }
            }
    /**********/
            column.find('.ui-cell-editor-output').hide();
            column.find('.ui-cell-editor-input').show();
        });
    },
```

Best Regards,

**Comment by christophs78** [comment]
@mertsincan: Thanks for your valuable feedback.

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
Good teamwork on this PR. :)

**Comment by tandraschko** [comment]
is the feedback from mert about lazy already implemented?

**Comment by christophs78** [comment]
Yes, it is. (https://github.com/primefaces/primefaces/pull/5233/commits/b03082954ede0e4552554e34a6bf3a16f63633cc)

**Pull request merged**